src/uu/install/src/install.rs

```diff
@@ -999,6 +999,48 @@ fn copy(from: &Path, to: &Path, b: &Behavior) -> UResult<()> {
     Ok(())
 }
 
+/// Check if a file needs to be copied due to ownership differences when no explicit group is specified.
+/// Returns true if the destination file's ownership would differ from what it should be after installation.
+#[cfg(not(target_os = "windows"))]
+fn needs_copy_for_ownership(to: &Path, to_meta: &fs::Metadata) -> UResult<bool> {
+    use std::os::unix::fs::MetadataExt;
+
+    // Check if the destination file's owner differs from the effective user ID
+    if to_meta.uid() != geteuid() {
+        return Ok(true);
+    }
+
+    // For group, we need to determine what the group would be after installation
+    // If no group is specified, the behavior depends on the directory:
+    // - If the directory has setgid bit, the file inherits the directory's group
+    // - Otherwise, the file gets the user's effective group
+    let expected_gid = if let Some(parent) = to.parent() {
+        if let Ok(parent_meta) = metadata(parent) {
+            // Check if parent directory has setgid bit set
+            if parent_meta.mode() & 0o2000 != 0 {
+                // Directory has setgid, file will inherit directory's group
+                parent_meta.gid()
+            } else {
+                // No setgid, file gets user's effective group
+                getegid()
+            }
+        } else {
+            // Can't read parent metadata, assume user's effective group
+            getegid()
+        }
+    } else {
+        // No parent (shouldn't happen in practice), assume user's effective group
+        getegid()
+    };
```

Comment:
There's a lot of nesting ;-) I think you can improve it with something like:
```suggestion
    let expected_gid = to
        .parent()
        .and_then(|parent| metadata(parent).ok())
        .filter(|parent_meta| parent_meta.mode() & 0o2000 != 0)
        .map(|parent_meta| parent_meta.gid())
        .unwrap_or(getegid());
```

Reply:
wahou, i am impressed :)